generative_ai/prompts/prompt_optimizer.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+
+#  [START generativeaionvertexai_prompt_optimizer]
+def optimize_prompts(
+    project: str,
+    location: str,
+    staging_bucket: str,
+    configuration_path: str,
+) -> str:
+    """Improve prompts by evaluating the model's response to sample prompts against specified evaluation metric(s).
+    Args:
+        project: Google Cloud Project ID.
+        location: Location where you want to run the Vertex AI prompt optimizer.
+        staging_bucket: Specify the Google Cloud Storage bucket to store outputs and metadata. For example, gs://bucket-name
+        configuration_path: URI of the configuration file in your Google Cloud Storage bucket. For example, gs://bucket-name/configuration.json.
+    Returns:
+        custom_job.resource_name: Returns the resource name of the job created of type: projects/project-id/locations/location/customJobs/job-id
+    """
+    from google.cloud import aiplatform
+
+    aiplatform.init(project=project, location=location, staging_bucket=staging_bucket)
```

Comment:
fix: Please add a section with variables that should be updated and commented our by the user ([example](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/9aa31c1e1defc09f1392fe09db4332221fb68bbe/generative_ai/function_calling/chat_function_calling_basic.py#L27))